### PR TITLE
feat: Remove dependency on deprecated `hashicorp/template` provider

### LINKED ISSUE #1294: Deprecated "registry.terraform.io/hashicorp/template"
## Description

Terraform eks 14 module is using deprecated "template provider", which doesn't support mac m1.  Ref. https://github.com/hashicorp/terraform-provider-template
Error while running `terraform init`
```
Incompatible provider version

Provider registry.terraform.io/hashicorp/template v2.2.0 does not have a
package available for your current platform, darwin_arm64.
```
## Versions

 - Terraform: v0.14.10

 - Provider(s): terraform-aws-modules/eks/aws

 - Module: 14.0

## Reproduction

Run terraform module terraform-aws-modules/eks/aws with version 14 on Mac M1.

### Code Snippet to Reproduce

image

### PR DESCRIPTION
**Pull request by dmsalomon**
# PR o'clock

## Description

hashicorp/template is deprecated, and the templatefile function should be used instead. In particular, this is problem on new Apple silicon Macs, since the hashicorp/template provider is not implemented (and I don't ever will be) for darwin_arm64. 

Resolves #1294

I have replaced the "template_file" data sources with local variables that store the outputs to function calls using `template`. There is actually very little that has changed since the resource arguments map cleanly to the function arguments. I had to use an explicit for loop to render the templates for every node, instead of using the `count` meta-argument.

I'm relatively new to Terraform, so all critiques are appreciated.

### Checklist

- [ ] CI tests are passing
- [ ] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by barryib** [comment]
This break some use cases. For the record, see https://github.com/hashicorp/terraform/pull/24978

**Comment by barryib** [comment]
@dmsalomon Thanks for working on this. Can you please upgrade your branch ?

**Comment by dmsalomon** [comment]
@barryib Thanks for taking a look at this, I just upgraded the branch.

File: versions.tf [hunk 1]
```diff
@@ -5,7 +5,6 @@ terraform {
     aws        = ">= 3.35.0"
     local      = ">= 1.4"
     null       = ">= 2.1"
-    template   = ">= 2.1"
```
Comment (barryib) [inline versions.tf:8]: Can you please also remove the template provider from required providers in [examples](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples).

**Review by barryib — CHANGES_REQUESTED** [review]

**Review by barryib — APPROVED** [review]

**Comment by barryib** [comment]
Thanks @dmsalomon for your contribution.

**Comment by barryib** [comment]
This now shipped in [v16.0.0](https://github.com/terraform-aws-modules/terraform-aws-eks/releases/tag/v16.0.0)

**Pull request merged**